histogram.go

```diff
@@ -151,8 +151,15 @@ func (m *histogram) Observe(val float64) {
 
 	m.sum += val
 	m.count++
-	b := &m.buckets[m.search(val)]
-	b.count++
+
+	// Buckets MUST be cumulative.
+	// As an example for a metric representing request latency in seconds its values for buckets with thresholds 1, 2, 3, and +Inf MUST follow value_1 <= value_2 <= value_3 <= value_+Inf.
+	// If ten requests took 1 second each, the values of the 1, 2, 3, and +Inf buckets MUST equal 10.
+	bi := m.search(val)
+	for i := 0; i <= bi; i++ {
+		b := &m.buckets[i]
+		b.count++
+	}
 }
 
 func (m *histogram) ObserveExemplar(ex *Exemplar) {
```

Comment:
you need to update this func too + test pls

Reply:
Will update; regarding test - we already have one (explicit call for ObserveExemplar + asserting that appended points contain that exemplar on matching bucket/point).